Applications/DataExplorer/DataView/NetCdfConfigureDialog.cpp

```diff
@@ -162,63 +150,67 @@ void NetCdfConfigureDialog::on_comboBoxDim4_currentIndexChanged(int id)
 	}
 }
 
-void NetCdfConfigureDialog::setVariableSelect()
+int NetCdfConfigureDialog::setVariableSelect()
 {
-	for (int i=0; i<(_currentFile->num_vars()); i++)
+	std::size_t max_dim(0);
+	std::size_t max_dim_idx(0);
+	std::size_t const n_vars (_currentFile->num_vars());
+	for (int i=0; i<n_vars; i++)
 	{
-		NcVar *focusedVar = _currentFile->get_var(i);
-		if (focusedVar->num_dims() > 0) comboBoxVariable->addItem(focusedVar->name());
+		NcVar const& focusedVar = *_currentFile->get_var(i);
+		std::cout << focusedVar.num_dims() << std::endl;
+		if (focusedVar.num_dims() > 1)
+		{
+			_id_map.push_back(i);
+			comboBoxVariable->addItem(focusedVar.name());
+			if (focusedVar.num_dims() > max_dim)
+			{
+				max_dim = focusedVar.num_dims();
+				max_dim_idx = comboBoxVariable->count() - 1;
+			}
+		}
 	}
+	return max_dim_idx;
 }
 
 void NetCdfConfigureDialog::setDimensionSelect()
 {
-	comboBoxDim1->clear();
-	comboBoxDim2->clear();
-	comboBoxDim3->clear();
-	comboBoxDim4->clear();
-	for (int i=0; i < _currentVar->num_dims(); i++) //write dimension-names into selection-boxes
+	int const dim (_currentVar->num_dims());
+	std::array<QComboBox*,4> dim_box = { comboBoxDim1, comboBoxDim2, comboBoxDim3, comboBoxDim4 };
+
+	for (std::size_t i=0; i<4; ++i)
 	{
-		comboBoxDim1->addItem(_currentVar->get_dim(i)->name());
-		comboBoxDim2->addItem(_currentVar->get_dim(i)->name());
-		comboBoxDim3->addItem(_currentVar->get_dim(i)->name());
-		comboBoxDim4->addItem(_currentVar->get_dim(i)->name());
+		dim_box[i]->clear();
+		dim_box[i]->setEnabled(i < dim);
 	}
-	if (_currentVar->num_dims() < 4)
+
+	for (int i=0; i < dim; ++i) //write dimension-names into selection-boxes
 	{
-		comboBoxDim4->setEnabled(false);comboBoxDim4->clear();
-		spinBoxDim4->setEnabled(false);spinBoxDim4->setValue(0);
-	}else{  //pre-set dimension selection, typical for 4-d-nc-files:
-		comboBoxDim4->setEnabled(true);
-		spinBoxDim4->setEnabled(true);
-		comboBoxDim1->setCurrentIndex(2);on_comboBoxDim1_currentIndexChanged(2);
-		comboBoxDim2->setCurrentIndex(3);on_comboBoxDim2_currentIndexChanged(3);
-		comboBoxDim3->setCurrentIndex(0);on_comboBoxDim3_currentIndexChanged(0);
-		comboBoxDim4->setCurrentIndex(1);on_comboBoxDim4_currentIndexChanged(1);
+		for (std::size_t j=0; j<_currentVar->num_dims(), j<4; ++j)
+			dim_box[j]->addItem(_currentVar->get_dim(i)->name());
```

Comment:
each variable in the file has a number of dimension between 0 and 4. we will only import variables with dim>=2. Variables with dim<2 are only scalars, scales, ranges or other support data. the names of the dimensions of each variable have to be inserted into the combobox menus in the dialogue, that is what is happening here. I have 4 comboboxes, not all of which have necessarily associated variables.  So I'm looping over these boxes and fill in the possible variable names for each dimension. The limit for i and j is the same, with a maximum of 4.